qa/restart-upgrade/build.gradle

```diff
@@ -65,12 +65,21 @@ task testAgainstOldCluster(type: StandaloneRestIntegTestTask) {
     systemProperty 'tests.skip_delete_model_index', 'true'
     systemProperty 'tests.plugin_bwc_version', ext.neural_search_bwc_version
 
-    //Excluding MultiModalSearchIT, HybridSearchIT, NeuralSparseSearchIT tests from neural search version 2.9 and 2.10 because these features were released in 2.11 version.
+    //Excluding MultiModalSearchIT, HybridSearchIT, NeuralSparseSearchIT, NeuralQueryEnricherProcessorIT tests from neural search version 2.9 and 2.10
+    // because these features were released in 2.11 version.
     if (ext.neural_search_bwc_version.startsWith("2.9") || ext.neural_search_bwc_version.startsWith("2.10")){
         filter {
             excludeTestsMatching "org.opensearch.neuralsearch.bwc.MultiModalSearchIT.*"
             excludeTestsMatching "org.opensearch.neuralsearch.bwc.HybridSearchIT.*"
             excludeTestsMatching "org.opensearch.neuralsearch.bwc.NeuralSparseSearchIT.*"
+            excludeTestsMatching "org.opensearch.neuralsearch.bwc.NeuralQueryEnricherProcessorIT.*"
+        }
+    }
+
+    // Excluding the test because we introduce this feature in 2.13
+    if (ext.neural_search_bwc_version.startsWith("2.11") || ext.neural_search_bwc_version.startsWith("2.12")){
```

Comment:
There are 3 cases

1) bwc version is 2.11 and current version is 2.12
For this the check should be
if (ext.neural_search_bwc_version.startsWith("2.11")  && ext.opensearch_version.startsWith("2.12")){
something on the above lines

2) bwc version is 2.11 and current version is on or after 2.13
This condition is not needed because processor will be created in the older version and search will executed in the newer version


3) bwc version is 2.13 and current version is on after 2.14

Reply:
For case 1 I think we can ignore because the codes in this PR won't be in 2.12.

for case 2, do you mean we don't need to do search in old cluster? This will introduce a behavior different in old cluster( if bwcVersion before 2.13 we just create processor, if after 2.13, we also do search). And we'll need to add bwcVersion check in the IT, like the code I removed yesterday.

Reply:
Got it.